### ISSUE TITLE
Tooltips and scrolling

### ISSUE THREAD

**Issue by pliablepixels** [tomsouthall/Chart.HorizontalBar.js#3]
Hello, I am using your horizontal chart for my project and am facing the following problems:

I installed the module via:

```
bower install chart.horizontalbar
```

I am using tc-chart-js which wraps Chart.js with angular directives

a) Tooltips don't work -- I have one tab where I am using horizontal tabs with your plugin and another tab where I am using regular vertical bar charts of Chart.js. The source of the data for both is the same.

b) Would you have any idea how to control the height? This is not really a problem with your code, its inherent in Chart.js but you might know enough to help. What happens is that all bars are scrunched into the same screen. What I really want is for it to scroll if bar height gets less than, say, 30 pixels. I've been tinkering with wrapping the canvas in overflow:scroll div, but while that works with many bars, when there are few bars, chart.js expands their height to make it scroll.

The invoking code:

```
<canvas tc-chartjs chart-type="HorizontalBar" chart-data="data" chart-options="options" >
                        </canvas>
```

thanks

**Comment by berturion** [comment]
I confirm that Tooltips doesn't work. Anyway, having this horizontal bar is great. Thanks.

**Comment by marulango** [comment]
+1

**Comment by dannzen** [comment]
Change following Code in Line 80 

```
inRange : function(chartX,chartY){
    return (chartY >= this.y - this.height/2 && chartY <= this.y + this.height/2) && (chartX >= this.left && chartX <= this.right);
}
```

to

```
inRange : function(chartX,chartY){
    return (chartY >= this.y - this.height/2 && chartY <= this.y + this.height/2) && (chartX >= this.left);
}
```

 and its showing the tooltips this.right is undefined and i couldnt find the cause, but this is solving my problem

**Comment by jakenoble** [comment]
That didn't seem to work for me. My Tool Tips are way off on the right hand side.

http://cl.ly/image/2Q2z0V3x1E1E

**Comment by dannzen** [comment]
the tooltips are  shown at the middle/end of the bar, it seems your bars are ending outside of it.

**Comment by jakenoble** [comment]
Yeah that does seem to be the case - any ideas why that might be happening?

Its just a regular canvas with a height and width in CSS and on the element.

**Comment by dannzen** [comment]
Try removing the height and width config and let it auto detected the scaling. Never had that problem :(

**Comment by jakenoble** [comment]
Perfect! Worked!

**Comment by pliablepixels** [comment]
I'm not using this plugin anymore but  since the solution suggested works I'll close this.

**Comment by Aaronm14** [comment]
This is technically still an issue.  It looks like this got fixed in a way in PR #2 although this fix still doesn't show tool tips for items that have a value of 0, whereas the fix in this thread does.  However, the PR code isn't in the version that bower serves up.  My thought would be that the best solution would be using the code @lieferant4me wrote and then making sure to update bower.json so the fix gets included with the source.
